src/inquirer/render/console/_list.py

```diff
@@ -90,6 +94,18 @@ def process_input(self, pressed):
 
             raise errors.EndOfInput(getattr(value, "value", value))
 
+        if self.question.matcher is not None:
+            if pressed.isprintable():
+                self.input += pressed
+                for choice in self.question.choices:
+                    if self.question.matcher(choice, self.input):
+                        self.current = self.question.choices.index(choice)
+                        break
+
+            # BACKSPACE gives a \x7f on linux while it gives a \x08 on windows
+            if pressed == chr(127) or pressed == chr(8):
```

Comment:
This is what the `readchar.key` module is for, it holds the OS specific key-codes.

```suggestion
            if pressed == key.BACKSPACE:
```